src/main/java/hudson/plugins/jira/JiraCreateIssueNotifier.java

```diff
@@ -305,35 +300,30 @@ private void writeInFile(String Filename, Issue issue) throws FileNotFoundExcept
      * It adds comment until the previously created issue is closed.
      */
     private void currentBuildResultFailure(AbstractBuild<?, ?> build, BuildListener listener, Result previousBuildResult,
-                                           String filename, EnvVars environmentVariable) throws InterruptedException, IOException {
+                                           String filename, EnvVars vars) throws InterruptedException, IOException {
 
-        String buildURL = environmentVariable.get("BUILD_URL");
-        String buildNumber = environmentVariable.get("BUILD_NUMBER");
         if (previousBuildResult == Result.FAILURE) {
-            String comment = String.format("- Job is still failing.\n- Failed run : [%s|%s]\n** [console log|%s]",
-                    buildNumber, buildURL, buildURL.concat("console"));
+            String comment = String.format("Build is still failing.\nFailed run: %s", getBuildDetailsString(vars));
+
             //Get the issue-id which was filed when the previous built failed
             String issueId = getIssue(filename);
             if (issueId != null) {
-                listener.getLogger().println("*************************Test fails again******************************");
                 try {
                     //The status of the issue which was filed when the previous build failed
-                    String Status = getStatus(build, issueId);
+                    Status status = getStatus(build, issueId);
 
-                    //Status=1=Open OR Status=5=Resolved
-                    if (Status.equals("1") || Status.equals("5")) {
-                        listener.getLogger().println("The previous build also failed creating issue with issue ID " + issueId);
-                        addComment(build, issueId, comment);
-                    }
-
-                    if (Status.equals("6")) {
+                    // Issue Closed, need to open new one
+                    if (status.getName().equals("Closed") || status.getName().equals("Done")) {
```

Comment:
since statuses are pre-defined constants, it will be better to store them in Enum object